thorlcr/graph/thgraph.hpp

```diff
@@ -583,9 +584,14 @@ class graph_decl CGraphBase : public CInterface, implements IEclGraphResults, im
     CGraphBase *queryOwner() { return owner; }
     CGraphBase *queryParent() { return parent?parent:this; }
     IMPServer &queryMPServer() const;
+    void clearProgressUpdated() { progressUpdated.store(false); }
+    void setProgressUpdated() { progressUpdated.store(true); }
+    bool hasProgressUpdated()
+    {
+        bool b=true;
+        return progressUpdated.compare_exchange_strong(b, false);
```

Comment:
I think progressUpdated.exchange(false); is more efficient, and also clearer in its intent.